### ISSUE TITLE
SyntaxError: invalid escape sequence due to "\s" found in skorch.dataset docstring

### ISSUE THREAD

**Issue by RomanBredehoft** [skorch-dev/skorch#999]
Hello ! 
We recently encounter an issue where importing Skorch leads to a `SyntaxError: invalid escape sequence due to "\s"` in the [dataset.py](https://github.com/skorch-dev/skorch/blob/master/skorch/dataset.py) file. We installed version 0.11.0 with poetry but also tried until 0.14.0 and still encountered the issue. 

After looking at the source code, it indeed seems that the docstring for the Dataset class has [some "\s" ](https://github.com/skorch-dev/skorch/blob/e83e6a4d8f26b96c1f0fc6931096219bf4f2628f/skorch/dataset.py#L121). I also can see that there is a `# pylint: disable=anomalous-backslash-in-string` at the beginning of it :
```
class Dataset(torch.utils.data.Dataset):
    # pylint: disable=anomalous-backslash-in-string
    """General dataset wrapper that can be used in conjunction with
    PyTorch :class:`~torch.utils.data.DataLoader`.

    The dataset will always yield a tuple of two values, the first
    from the data (``X``) and the second from the target (``y``).
    However, the target is allowed to be ``None``. In that case,
    :class:`.Dataset` will currently return a dummy tensor, since
    :class:`~torch.utils.data.DataLoader` does not work with
    ``None``\s.

    :class:`.Dataset` currently works with the following data types:

    * numpy ``array``\s
    * PyTorch :class:`~torch.Tensor`\s
    * scipy sparse CSR matrices
    * pandas NDFrame
    * a dictionary of the former three
    * a list/tuple of the former three
```

Is there a reason why these "\s" are found here ? I've looked a bit into other files and could not manage to stumble on other examples. It seems that it is specific to this particular docstring. 

In fact, I believe they should be either removed (or at least doubled). As [PEP 257](https://peps.python.org/pep-0257/#what-is-a-docstring) states :
> For consistency, always use """triple double quotes""" around docstrings. Use r"""raw triple double quotes""" if you use any backslashes in your docstrings. For Unicode docstrings, use u"""Unicode triple-quoted strings""".

And I don't think this doscstring is meant to be a raw string literal !

Thanks !

**Comment by BenjaminBossan** [comment]
Hey, IIRC the reason was pretty profane, when the docstring is interpreted as RST for the docs, without the backslash, the preceding word would not be shown in monospace font, instead just displaying the backtick. I'm okay with the PR you submitted, having the plural form is not crucial here.

Before proceeding, I have to wonder though how you get the syntax error. If this was a general problem, I would expect more people to complain. Could you please tell me the conditions that trigger the error? What Python version are you using, what code triggers the error?

**Comment by RomanBredehoft** [comment]
Hello @BenjaminBossan , sorry for the late answer !
Ok that is indeed .. unexpected 😅 But yes thanks for pointing that out, I surely wouldn't want to revert some things like this even as negligible as it can be. 

So basically my error is also very odd. After some diggings, it seems that it comes from [Poetry](https://python-poetry.org/) it self (with version 1.3 and above) and the way the install libraries with backslashes in some of their docstrings. 

Besides, I tried to have a fresh and simple virtual env with only Poetry and Skorch, but I unfortunately was not able to reproduce it as is. For now, the only way I'm able to do so is using a much more complex environment (here's the [lock](https://github.com/zama-ai/concrete-ml/blob/main/pyproject.toml) file from our project), which is even weirder and does not help me a lot on finding that out. I'll get on it again next week and hopefully find a simple reproducible case. Then, I'll open a similar issue on Poetry's side and see what they say.

In the mean time, we found another issue on another repo which seems to have fixed the problem by making the docstrings "raw" (as PEP is suggesting actually) : https://github.com/pymc-devs/pymc/issues/3878. But I believe that might not be the best solution, and clearly this is not linked to Python 3.7 only (I'm using `Python 3.10.11`). Actually what I want to try is to instead double the backslashes (`\s` -> `\\s`) and see if that fixes the issue. But then, I'm not sure how this is going to be handled in the docs. Is there a simple way I could check the docs after making my change and make sure fonts stay the same ?

Many thanks !

**Comment by BenjaminBossan** [comment]
Very surprising indeed. Thanks a lot for digging really deep into the issue. If you could find a minimal env that reproduces the issue, that would be great.

As I mentioned, I will gladly accept the PR, even if it turned out at the end that it's a poetry issue, since the change is minimal and could help other people. I just feel more comfortable knowing the cause of the issue before taking action.

> Is there a simple way I could check the docs after making my change and make sure fonts stay the same ?

Yes, if you go into the `docs/` directory, you can call `make html` and then open `_build/html/index.html` in a browser and navigate > skorch API > skorch.dataset. The requirements for building the docs are in `docs/requirements.txt`.

**Comment by RomanBredehoft** [comment]
Great thanks for you quick answer ! I'll get back to you when I know more about it 😉

**Comment by BenjaminBossan** [comment]
Did you have time to investigate deeper?

**Comment by RomanBredehoft** [comment]
Unfortunately no not yet but I'm planning on soon, sorry for the delay !

**Comment by BenjaminBossan** [comment]
What a weird error, thanks so much @RomanBredehoft for your thorough investigation. Getting to the bottom of this issue, with how it somehow involves poetry and pytest, and how the import solves it, would probably result in a very interesting conference talk and important lesson. But let's not throw away more time if there is a solution at the ready.

> * use raw triple quotes for docstrings, which is recommended by pep

This looks like the way to go. Thanks for updating the PR.

**Comment by BenjaminBossan** [comment]
Fixed via #1000.
